src/dataflow/src/source/timestamp.rs

```diff
@@ -122,10 +126,20 @@ impl TimestampProposer {
     }
 }
 
-/// This struct holds per partition timestamp binding state, as a ordered list of bindings (time, offset).
-/// Each binding indicates "all offsets < offset must be bound to time", and adjacent pairs of bindings
-/// (time1, offset1), (time2, offset2) denote that offsets in [offset1, offset2) should get bound
-/// to time1.
+/// This struct holds per partition timestamp binding state, as an ordered list
+/// of pairs (time, offset). Each pair indicates "all offsets < offset must be
+/// bound to time".
```

Comment:
I think this text is incorrect, in that `(time, offset)` only requires that "all offsets < offset must be bound to a time that is at most `time`". A lesser offset might be bound to something less than `time`.

Reply:
thank you! Corrected.